File: atcoder/abc/abc313/c/main.cpp
#include <bits/stdc++.h>
using namespace std;


int main(){
  int N;
  vector<long> A(N);
  long v=0;
  for(int i=0;i<N;i++){
    cin >> A[i];
    v+=A[i];
  }
  if(v%N==0){
    long ave = v/N;
    long p=0;
    long m=0;
    for(auto a:A){
      if(a>ave) p+=ave-a;
      else m+=a-ave;
    }
    cout << (max(m, p)+1)/2 << endl;
  }
  else{
    long p = 0;
    long m = 0;
    long ave = v/N;
    long avee = ave+1;
    for(auto a: A){
      if(a>=ave) p+=min(a-ave, a-avee);
      else m+=min(ave-a, avee-a);
    }
    cout << max(p, m) << endl;
  }
  
}

File: atcoder/abc/abc313/c/main.py
N = int(input())
A = [int(i) for i in input().split()]

def calc(low, high):
  p = 0
  m = 0
  for a in A:
    if a==low or a==high: continue
    elif a>high: p += a-high
    else: m+=low-a
  
  return max(p, m)


v = sum(A)
ave = v//N
if v%N==0: print(min(calc(ave-1, ave), calc(ave,ave+1)))
else: print(calc(ave, ave+1))
